### PR TITLE
Follow API rate limit to avoid IP ban

### PR DESCRIPTION
**Pull request by FooIbar**
> Load limiting: 25 entries per request, 4-5 sequential requests usually okay before having to wait for ~5 seconds

https://ehwiki.org/wiki/API#Basics

### CODE REVIEW

**Review by VDostoyevskiy — APPROVED** [review]

File: app/src/main/kotlin/com/hippo/ehviewer/client/EhEngine.kt [hunk 1]
```diff
@@ -446,7 +445,10 @@ object EhEngine {
     }
 
     suspend fun addFavorites(galleryList: List<Pair<Long, String>>, dstCat: Int) {
-        galleryList.parMap(concurrency = Settings.multiThreadDownload) { (gid, token) ->
+        galleryList.forEachIndexed { index, (gid, token) ->
```
Comment (VDostoyevskiy) [inline app/src/main/kotlin/com/hippo/ehviewer/client/EhEngine.kt:448]: Do we actually need to sequentialize this?

Comment (FooIbar) [reply app/src/main/kotlin/com/hippo/ehviewer/client/EhEngine.kt:448]: Yeah I was thinking.
Delay on every request feels too much.

Comment (VDostoyevskiy) [reply app/src/main/kotlin/com/hippo/ehviewer/client/EhEngine.kt:448]: I mean, anyone really being IP banned due to this?

Comment (FooIbar) [reply app/src/main/kotlin/com/hippo/ehviewer/client/EhEngine.kt:448]: If someone has like 1k local favorites and tries moving them to cloud, he can easily get banned.

**Pull request merged**